samples: application_development: code_relocation_nocopy: sample.application_development.code_relocation_nocopy fails

**Describe the bug**
The test `sample.application_development.code_relocation_nocopy` fails.

Observed for `nrf5340dk/nrf5340/cpuapp`.

**To Reproduce**
Steps to reproduce the behavior:
1. have `nrf5340dk/nrf5340/cpuapp` connected
2. go to your zephyr dir
3. call `./scripts/twister -T samples/application_development/code_relocation_nocopy -p nrf5340dk/nrf5340/cpuapp --device-testing --device-serial /dev/ttyACM1 -v --inline-logs`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
```
/toolchains/d738e9cb55/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a(reset.S.obj): in function `__start':
zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp/tests/net/mld/net.mld/../../../../../../zephyr/arch/arm/core/cortex_m/reset.S:97: undefined reference to `z_arm_platform_init'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Toolchain Zephyr SDK 0.16.5
 - Commit SHA or Version used: `v3.6.0-2418-gb573f447f04f`

--- a/soc/nordic/CMakeLists.txt
+++ b/soc/nordic/CMakeLists.txt
@@ -2,7 +2,9 @@
 
 zephyr_library()
 
-set(SOC_LINKER_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/linker.ld CACHE INTERNAL "SoC Linker script")
+if(CONFIG_ARM)
+  set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/arm/cortex_m/scripts/linker.ld CACHE INTERNAL "SoC Linker script")
+endif()
 
 zephyr_library_sources(
   validate_base_addresses.c

--- a/soc/nordic/common/CMakeLists.txt
+++ b/soc/nordic/common/CMakeLists.txt
@@ -3,6 +3,8 @@
 
 add_subdirectory_ifdef(CONFIG_RISCV_CORE_NORDIC_VPR vpr)
 
+zephyr_linker_sources_ifdef(CONFIG_ARM SECTIONS arm_platform_init.ld)
+
 zephyr_library_sources_ifdef(CONFIG_POWEROFF poweroff.c)
 
 zephyr_include_directories(.)

--- a/soc/nordic/common/arm_platform_init.ld
+++ b/soc/nordic/common/arm_platform_init.ld
@@ -4,16 +4,5 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#if defined(CONFIG_ARM)
-#include <zephyr/arch/arm/cortex_m/scripts/linker.ld>
-
 /* Let SystemInit() be called in place of z_arm_platform_init() by default. */
 PROVIDE(z_arm_platform_init = SystemInit);
-
-#elif defined(CONFIG_RISCV)
-#include <zephyr/arch/riscv/common/linker.ld>
-
-#else
-#error Unsupported architecture
-
-#endif

--- a/soc/nordic/common/vpr/CMakeLists.txt
+++ b/soc/nordic/common/vpr/CMakeLists.txt
@@ -4,3 +4,5 @@
 zephyr_include_directories(.)
 
 zephyr_library_sources(soc_idle.c soc_irq.S soc_irq.c vector.S)
+
+set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/riscv/common/linker.ld CACHE INTERNAL "")

